go/ai/generate.go

```diff
@@ -1058,3 +1071,146 @@ func handleResumeOption(ctx context.Context, r *registry.Registry, genOpts *Gene
 		toolMessage: toolMessage,
 	}, nil
 }
+
+// DefineBackgroundModel defines a new model that runs in the background
+func DefineBackgroundModel(
+	r *registry.Registry,
+	provider, name string,
+	opts *BackgroundModelOptions[*ModelRequest, *ModelResponse],
+) BackgroundModel {
+	label := opts.Label
+	if label == "" {
+		label = name
+	}
+	metadata := map[string]any{
+		"model": map[string]any{
+			"label":    label,
+			"versions": opts.Versions,
+			"supports": opts.Supports,
+		},
+	}
+
+	if opts.ConfigSchema != nil {
+		metadata["model"].(map[string]any)["customOptions"] = opts.ConfigSchema
```

Comment:
This can panic, do a check on whether it cast correctly:

`if meta, ok := metadata["model"].(map[string]any); ok {`